src/Kernel/ThisProcessVariable.class.st

```diff
@@ -0,0 +1,24 @@
+"
+I model thisProcess keyword
+"
+Class {
+	#name : #ThisProcessVariable,
+	#superclass : #ReservedVariable,
+	#category : #'Kernel-Variables'
+}
+
+{ #category : #accessing }
+ThisProcessVariable class >> variableName [
+	^#thisProcess
+]
+
+{ #category : #testing }
+ThisProcessVariable >> isThisProcessVariable [
+	^true
+]
+
+{ #category : #debugging }
+ThisProcessVariable >> readInContext: aContext [
+	"We expect this method to be called either from the same process or by debugger process that will read the correct process. Otherwise we cannot guarantee to read the correct process"
+	^ Processor activeProcess
```

Comment:
This one is problematic. I do not like the idea to silently return possible a bad value. I fear that this will bite people and cause inconsistencies.

I prefer to have an active runtime error saying that `thisProcess` in unreliable, or a passive error (return nil) instead of having a bad value.

I'm not sure how exactly contexts and processes are associated. If a context can be dead and associated to no context, returning nil might make sense.
If a context can be associated to a single process at most, searching the correct process can be a solution.
I prefer a slow correct answer to a fast wrong one.

Reply:
When exploring the senders, we saw that it is not used outside of the debugger, so we don't expect this to be a problem. But returning nil could also be a solution to be sure we have a consistent output.